mailbox/noise.go

```diff
@@ -501,18 +509,14 @@ const (
 	// 1 + 33 + 16
 	ActOneSize = 50
 
-	// ActTwoPayloadSize is the size of the fixed sized payload that can be
-	// sent from the responder to the initiator in act two.
-	ActTwoPayloadSize = 500
```

Comment:
Picked this initially as it seemed like one of those "good enough" values. There's also the added simplicity in keeping this fixed size as well. 

Do we see a near term future that results in the macaroon being more than 500 bytes?

Also if we want to decouple things further, then what we can do is actually just send the final payload over in a message from the server to client, but consider this to be a final act (4) of the handshake.

Reply:
The other advantage of keeping this fixed size (even if we increase to say 1 KB or something) is traffic analysis: all handshakes look the same, and no information can be inferred w.r.t the size of the macaroon payload in the second act.

Reply:
> Do we see a near term future that results in the macaroon being more than 500 bytes?

The current admin supermacaroon is already more than 500 bytes. Was busy with the admin-macaroon session task and then notices that the macaroon was getting cut in half here which is why i had to make this change.

> Also if we want to decouple things further, then what we can do is actually just send the final payload over in a message from the server to client, but consider this to be a final act (4) of the handshake.

yeah could do this and then in act 2 just send over the size of the payload to be expected in act 4. this would then also allow us to get rid of the previous commit since we will always know how many bytes the next message will be. However, this still has the issue of traffic analysis....

Reply:
The current admin supermacaroon we bake is 1026 bytes

Reply:
> However, this still has the issue of traffic analysis....

Yeah the main way to get around this would be to pick something "large enough" and always send that, trading off the extra bandwidth for the privacy gain. So it could be like 65 KB each time or something.